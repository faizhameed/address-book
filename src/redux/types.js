export const actionTypes = {
  ADD_BOOK: "ADD_BOOK",
};
