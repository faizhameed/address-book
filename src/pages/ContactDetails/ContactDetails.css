/* No CSS *//*# sourceMappingURL=ContactDetails.css.map */