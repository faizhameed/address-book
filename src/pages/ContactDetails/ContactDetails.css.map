{
    "version": 3,
    "mappings": "",
    "sources": [
        "ContactDetails.scss"
    ],
    "names": [],
    "file": "ContactDetails.css"
}