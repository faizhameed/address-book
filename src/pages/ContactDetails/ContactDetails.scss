.container {
  display: block;
  margin: 20px auto;
}
