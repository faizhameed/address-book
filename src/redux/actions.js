import { actionTypes } from "./types";
import Axios from "axios";

export const fetchContactList = (page = 1) => {
  let url = `https://randomuser.me/api/?${page}&results=50`;

  return function (dispatch) {
    dispatch({
      type: actionTypes.GET_CONTACTS_PENDING,
    });
    Axios.get(url)
      .then((response) =>
        dispatch({
          type: actionTypes.GET_CONTACTS_SUCCESS,
          payload: response.data,
        })
      )
      .catch((response) =>
        dispatch({
          type: actionTypes.GET_CONTACTS_FAILED,
          payload: response.error,
        })
      );
  };
};

export const updatePageNumber = (page) => ({
  type: actionTypes.UPDATE_PAGE_NUMBER,
  payload: page,
});

export const searchFieldChange = (data) => ({
  type: actionTypes.SEARCH_FIELD_CHANGE,
  payload: data,
});

export const updatePersonDetailView = (person) => ({
  type: actionTypes.UPDATE_PERSON_DETAIL,
  payload: person,
});

export const toggleDarkMode = () => ({
  type: actionTypes.TOGGLE_DARK_MODE,
});
