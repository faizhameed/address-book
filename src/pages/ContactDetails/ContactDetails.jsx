import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import ProfileCard from "../../components/Cards/Card";
import "./ContactDetails.scss";
import { Button } from "semantic-ui-react";
import Header from "../../components/Header/Header";

const ContactDetails = ({ person }) => {
  return (
    <React.Fragment>
      <Header content={`Profile`} />
      <div className="main">
        {person ? (
          <div className="container">
            <ProfileCard />
          </div>
        ) : (
          <div>
            <p>Sorry No ContactList selected</p>
          </div>
        )}
        {/**
         * Go back to address table
         */}
        <Link style={{ color: "#fff" }} to="/">
          <Button primary>Go Back</Button>
        </Link>
      </div>
    </React.Fragment>
  );
};

const mapStateToProps = ({ personDetailReducer: { person } }) => ({
  person,
});

export default connect(mapStateToProps)(ContactDetails);
