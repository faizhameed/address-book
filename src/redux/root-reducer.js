import { combineReducers } from "redux";
import { actionTypes } from "./types";

const initialStateContactList = {
  contactList: [],
  error: null,
  isPending: false,
  page: 1,
};
function contactListReducer(state = initialStateContactList, action) {
  switch (action.type) {
    case actionTypes.GET_CONTACTS_SUCCESS:
      return {
        ...state,
        contactList: [...state.contactList, ...action.payload.results],
        isPending: false,
      };

    case actionTypes.GET_CONTACTS_PENDING:
      return {
        ...state,
        isPending: true,
      };
    case actionTypes.GET_CONTACTS_FAILED:
      return {
        ...state,
        isPending: false,
        error: action.payload,
      };

    case actionTypes.UPDATE_PAGE_NUMBER:
      if (action.payload < 11) {
        return {
          ...state,
          page: action.payload,
        };
      } else {
        return state;
      }
    default:
      break;
  }
  return state;
}

const rootReducer = combineReducers({
  contactListReducer,
});

export default rootReducer;
