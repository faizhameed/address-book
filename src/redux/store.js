import { createStore } from "redux";
import rootReducer from "./root-reducer";
import { applyMiddleware } from "redux";
import { createLogger } from "redux-logger";
import thunkMiddleware from "redux-thunk";

const logger = createLogger();
export const store = createStore(
  rootReducer,
  applyMiddleware(thunkMiddleware, logger)
);
