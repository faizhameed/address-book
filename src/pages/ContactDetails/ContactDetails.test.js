import { shallow } from "enzyme";
import ContactDetails from "./ContactDetails";
import React from "react";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

describe("Profile Page  ", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Provider store={store}>
        <ContactDetails />
      </Provider>
    );
  });

  it("Profile Page is rendered Properly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
