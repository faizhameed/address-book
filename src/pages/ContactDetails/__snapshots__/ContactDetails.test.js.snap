// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page   Profile Page is rendered Properly 1`] = `ShallowWrapper {}`;
